File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.js
function solution(arr) {
    let pow_list = [1,2,4,8,16,32,64,128,256,512,1024,2048];
    let target_length = 0;
    
    for(len of pow_list) {
        if(len >= arr.length) {
            target_length = len;
            break;
        }
    }
    
    const initial_length = arr.length;
    for (let i = 0; i < target_length - initial_length; i++) {
        arr.push(0)
    }

    return arr;
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.py
def solution(arr):
    arr_len = len(arr)
    lengths = [1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024]
    
    if arr_len in lengths:
        return arr
    
    for leng in lengths:
        if arr_len < leng:
            arr_len = leng - arr_len
            break

    for add in range(arr_len):
        arr.append(0)

    return arr
    